### ISSUE TITLE
Weights in onnx->Halide

### ISSUE THREAD

**Issue by chtsao8** [halide/Halide#4050]
@asaadaldien Hi again,

I was wondering if pretrained weights are preserved during the conversion from Pytorch->onnx->Halide. Testing the alexnet model in PyTorch and in its converted Halide format don't yield the same results, and I'm not quite sure why.

Thanks!

**Comment by asaadaldien** [comment]
I am assuming you are running inference ? make sure you have in Pytorch model.eval() is probably set otherwise Dropout run in training mode and cause discrepancy.

**Comment by chtsao8** [comment]
Even with model.eval(), the converted Halide implementation of Alexnet still doesn't return a correct classification when testing. I actually suspect that it might have to do with my conversion of the image from uint8_t to float, though it seems fine - I used 3 for loops to cast each input pixel to a float and put them in a float buffer.

Additionally, I get dramatically different output vectors when running Alexnet in Pytorch vs Halide. In Pytorch, the values of the (1,1000) output vector are typically between 1-10, while in Halide they're between 1-200..

**Comment by benoitsteiner** [comment]
The implementation of the local response normalization isn't correct, which is why you're getting incorrect results. This will be fixed once https://github.com/halide/Halide/pull/4097 gets merged.

**Comment by steven-johnson** [comment]
Is this now fixed?

**Comment by benoitsteiner** [comment]
This is fixed for us internally, but I'd like to make sure it's fixed for @chtsao8 as well before closing the issue since we don't use the pytorch to onnx path but rely on the reference model shipped with the onnx compliance testsuite.

**Comment by chtsao8** [comment]
Do I need to update my entire Halide repo? Or just the modified files in the onnx app?
